src/cpu/matmul/gemm_bf16_matmul.cpp

```diff
@@ -145,7 +145,9 @@ status_t gemm_bf16_matmul_t<dst_type>::pd_t::check_and_configure_attributes(
     // set state
     CHECK(params_.pp_attr_.copy_from(*attr()));
     params_.gemm_applies_output_scales_
-            = attr()->scales_.get_mask(DNNL_ARG_WEIGHTS) == 0 && !with_bias();
+            = (attr()->scales_.has_default_values(DNNL_ARG_WEIGHTS)
+                      || attr()->scales_.get_mask(DNNL_ARG_WEIGHTS) == 0)
```

Comment:
Looks like it's a product of my change to scales pushed to v3.9 or maybe to v3.8 as well (a backport would be nice).

I also suggest to introduce a dedicated variable and the following logic:
```
    bool apply_wei_scales_in_pp_kernel = false;
    if (!attr()->scales_.has_default_values(DNNL_ARG_WEIGHTS))
        apply_wei_scales_in_pp_kernel  = attr()->scales_.get_mask(DNNL_ARG_WEIGHTS) > 0;

    ...
    params_.gemm_applies_output_scales_ = !apply_wei_scales_in_pp_kernel && !with_bias();
```